modules/rabbitmq/rabbitmq.go

```diff
@@ -0,0 +1,294 @@
+// See withExecutable.
+func WithPermission(p Permission) testcontainers.CustomizeRequestOption {
+	return withExecutable(p)
+}
+
+// WithPolicy declares the policy on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithPolicy(p Policy) testcontainers.CustomizeRequestOption {
+	return withExecutable(p)
+}
+
+// WithQueue declares the queue on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithQueue(q Queue) testcontainers.CustomizeRequestOption {
+	return withExecutable(q)
+}
+
+// WithVirtualHost declares the virtual host on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithVirtualHost(v VirtualHost) testcontainers.CustomizeRequestOption {
+	return withExecutable(v)
+}
+
+// WithVirtualHostLimit declares the virtual host limit on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithVirtualHostLimit(l VirtualHostLimit) testcontainers.CustomizeRequestOption {
+	return withExecutable(l)
+}
+
+// WithVirWithusertualHost declares the user on the RabbitMQ container, using "rabbitmqadmin".
+// See withExecutable.
+func WithUser(u User) testcontainers.CustomizeRequestOption {
+	return withExecutable(u)
+}
+
+// WithPluginsEnabled enables the specified plugins on the RabbitMQ container.
+// It will leverage the container lifecycle hooks to call "rabbitmq-plugins"
+// right after the container is started, enabling the plugins.
+func WithPluginsEnabled(plugins ...Plugin) testcontainers.CustomizeRequestOption {
+	ps := make([]Executable, len(plugins))
+	for i, p := range plugins {
+		ps[i] = p
+	}
+
+	return withExecutable(ps...)
+}
+
+// withExecutable will execute the command representation of the Executable into the container.
+// It will leverage the container lifecycle hooks to call a rabbitmq binary right after the container
+// is started.
+func withExecutable(execs ...Executable) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		for _, exec := range execs {
+			execFn := func(ctx context.Context, c testcontainers.Container) error {
+				_, _, err := c.Exec(ctx, exec.AsCommand())
+				return err
+			}
+
+			req.LifecycleHooks[0].PostStarts = append(req.LifecycleHooks[0].PostStarts, execFn)
+		}
+	}
+}
+
+// WithSSL enables SSL on the RabbitMQ container, adding the necessary environment variables,
+// files and waiting conditions.
+// From https://hub.docker.com/_/rabbitmq: "As of RabbitMQ 3.9, all of the docker-specific variables
+// listed below are deprecated and no longer used. Please use a configuration file instead;
+// visit https://rabbitmq.com/configure to learn more about the configuration file. For a starting point,
+// the 3.8 images will print out the config file it generated from supplied environment variables.
+func WithSSL(sslSettings SSLSettings) testcontainers.CustomizeRequestOption {
```

Comment:
take into account the following [PR comment](https://github.com/testcontainers/testcontainers-java/pull/7338#pullrequestreview-1555502237)

Reply:
I've not implemented yet the SSL improvements. Should I proceed with the current state or rework the PR to remove the usage of the deprecated variables. It will define the lower version of RabbitMQ to use: +3.9

Reply:
IMO new Testcontainers implementations should take into account the most generic and compatible approach. In this case, using the configuration file is an option which will work for versions older than 3.9 and new ones. The implementation will differ from Testcontainers for Java but I think it will be a good one too